### ISSUE TITLE
Diaspora: Empty links aren't rendered

### ISSUE THREAD

**Issue by Hypolite** [friendica/friendica#2993]
I just received a Diaspora item that contains an empty link still showing on [the source post on Diaspora*](https://framasphere.org/posts/07c61f109cbc0134ee8a2a0000053625).

Here's the display on [my Friendica server](https://friendica.mrpetovan.com/display/07c61f109cbc0134ee8a2a0000053625).

Here's the item XML object:
````xml
<XML>
  <post>
    <status_message>
      <diaspora_handle>kephasp@framasphere.org</diaspora_handle>
      <guid>07c61f109cbc0134ee8a2a0000053625</guid>
      <created_at>2016-12-05 01:56:58 UTC</created_at>
      <provider_display_name/>
      <raw_message>Ceci n'est **pas** une situation imaginaire&#xA0;!&#xD;
[](http://fibres-noires.blogspot.fr/2016/12/le-respect-de-la-parole-des-concerne-es.html)&#xD;
&#xD;
#x&#xE9;nophobie #f&#xE9;minisme #racisme #homophobie #lgbt</raw_message>
      <public>true</public>
    </status_message>
  </post>
</XML>
````
And the expected resulting body (line feeds added for clarity):
````
Ceci n'est [b]pas[/b] une situation imaginaire !
[url=http://fibres-noires.blogspot.fr/2016/12/le-respect-de-la-parole-des-concerne-es.html][/url]
#[url=https://friendica.mrpetovan.com/search?tag=x%C3%A9nophobie]xénophobie[/url]
#[url=https://friendica.mrpetovan.com/search?tag=f%C3%A9minisme]féminisme[/url]
#[url=https://friendica.mrpetovan.com/search?tag=racisme]racisme[/url]
#[url=https://friendica.mrpetovan.com/search?tag=homophobie]homophobie[/url]
#[url=https://friendica.mrpetovan.com/search?tag=lgbt]lgbt[/url]
````

And the produced rendered HTML (line feeds added for clarity):
````html
Ceci n'est <strong>pas</strong> une situation imaginaire !
<a href="http://fibres-noires.blogspot.fr/2016/12/le-respect-de-la-parole-des-concerne-es.html" target="_blank"></a>
<br>
<br>
#<a href="https://friendica.mrpetovan.com/search?tag=x%C3%A9nophobie" class="tag" title="x&eacute;nophobie">x&eacute;nophobie</a>
#<a href="https://friendica.mrpetovan.com/search?tag=f%C3%A9minisme" class="tag" title="f&eacute;minisme">f&eacute;minisme</a>
#<a href="https://friendica.mrpetovan.com/search?tag=racisme" class="tag" title="racisme">racisme</a>
#<a href="https://friendica.mrpetovan.com/search?tag=homophobie" class="tag" title="homophobie">homophobie</a>
#<a href="https://friendica.mrpetovan.com/search?tag=lgbt" class="tag" title="lgbt">lgbt</a>
````

This issue is a tracker for me to add an empty link filter to diaspora2bb to prevent links to disappear in translation.

**Comment by annando** [comment]
Where is there an empty link?

**Comment by Hypolite** [comment]
Here: `[](http://fibres-noires.blogspot.fr/2016/12/le-respect-de-la-parole-des-concerne-es.html)`

In theory it shouldn't appear anywhere but Diaspora* creates a OEmbed-like display for the first like of the post, whether it actually shows in the post or not. Friendica doesn't do the same, so the link completely disappear from the resulting post.

The solution is to add a filter to diaspora2bb to clean such empty links.

**Comment by annando** [comment]
I understand. Isn't the solution that the link detection in "add_page_info_to_body" needs to be improved? See here: https://github.com/friendica/friendica/blob/develop/include/items.php#L277

**Comment by Hypolite** [comment]
Ha, I found the issue with the current regex trying to match the `[url=]...[/url]` form: it excludes links that are preceded by the special character `!`.

The original XML message shows a carriage return character `&#xD;` (= `\r`), but it is stripped before being transformed into the BBCode-enabled body. I added new lines for clarity but in the item table row, everything is on the same line, with the `[url]` tag touching the exclamation mark of the previous line. We then assume it's a special link and ignore it when performing `add_page_info_to_body`.

The solution seems then not to strip all carriage returns character when receiving a Diaspora post.
